### PR TITLE
[MPS] Add support for max_unpool2d

### PR DESCRIPTION
**Pull request by ignaciosica**
- Add MaxUnpooling.mm
- Remove nn.functional.max_unpool2d room test_mps UNIMPLEMENTED_XFAILLIST

Add support for max_unpool2d #77764

### CODE REVIEW

File: aten/src/ATen/native/mps/operations/MaxUnpooling.mm [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+  MPSGraphTensor* inputTensor_ = nil;
+  MPSGraphTensor* indicesTensor_ = nil;
+  MPSGraphTensor* outputTensor_ = nil;
+};
+
+static std::string getCacheKey(const Tensor& input,
+                               const Tensor& indices,
+                               const ScalarType& dtype,
+                               const IntArrayRef output_size) {
+  return "max_unpooling2d_forward_mps:" + getTensorsStringKey({input, indices}) + "[" + getArrayRefString(output_size) +
```
Comment (p4sscode) [inline aten/src/ATen/native/mps/operations/MaxUnpooling.mm:21]: Hi, I'm new to contributing to this project and I don't really grasp yet what to include in cache key? Is output_size and type sufficient? Or is better to include input and indices tensor keys?

Comment (malfet) [reply aten/src/ATen/native/mps/operations/MaxUnpooling.mm:21]: Cache key should include input and output tensor sizes as well as dtypes. But `getTensorsStringKeys` is probably OK)

Comment (p4sscode) [reply aten/src/ATen/native/mps/operations/MaxUnpooling.mm:21]: Great, thanks!

File: aten/src/ATen/native/mps/operations/MaxUnpooling.mm [hunk 2]
```diff
@@ -0,0 +1,141 @@
+#define TORCH_ASSERT_ONLY_METHOD_OPERATORS
+#include <ATen/native/mps/MPSGraphVenturaOps.h>
+#include <ATen/native/mps/OperationUtils.h>
+
+#include <ATen/ops/max_unpool2d.h>
+#include <ATen/ops/max_unpool2d_native.h>
+
+namespace at::native {
+namespace mps {
```
Comment (malfet) [inline aten/src/ATen/native/mps/operations/MaxUnpooling.mm:9]: Q: why not use anonymous namespace here and get rid of all the `static` declarations?

File: aten/src/ATen/native/mps/operations/MaxUnpooling.mm [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+struct CachedGraph : public MPSCachedGraph {
+  CachedGraph(MPSGraph* graph) : MPSCachedGraph(graph) {}
+  MPSGraphTensor* inputTensor_ = nil;
+  MPSGraphTensor* indicesTensor_ = nil;
+  MPSGraphTensor* outputTensor_ = nil;
+};
+
+static std::string getCacheKey(const Tensor& input,
+                               const Tensor& indices,
+                               const ScalarType& dtype,
```
Comment (malfet) [inline aten/src/ATen/native/mps/operations/MaxUnpooling.mm:19]: Not needed here, is it?
```suggestion
```

File: aten/src/ATen/native/mps/operations/MaxUnpooling.mm (hunk 2 above)
Comment (p4sscode) [reply aten/src/ATen/native/mps/operations/MaxUnpooling.mm:9]: You are right, it is a better practice to use anonymous namespace

File: aten/src/ATen/native/mps/operations/MaxUnpooling.mm (hunk 3 above)
Comment (p4sscode) [reply aten/src/ATen/native/mps/operations/MaxUnpooling.mm]: No, outdated param

**Comment by ignaciosica** [comment]
Is there any work pending before merging this pr?

**Pull request closed without merging**